### PR TITLE
Add new lints from Dart 2.17

### LINKED ISSUE #55: Add new dart 2.17 lints
- new [`use_super_parameters`](https://dart-lang.github.io/linter/lints/use_super_parameters.html)
- new [`use_enums`](https://dart-lang.github.io/linter/lints/use_enums.html)
- new [`use_colored_box`](https://dart-lang.github.io/linter/lints/use_colored_box.html)
- marked stable [`null_check_on_nullable_type_parameter`](https://dart-lang.github.io/linter/lints/null_check_on_nullable_type_parameter.html)

### PR DESCRIPTION
**Pull request by bartekpacia**
close #55

### CODE REVIEW

**Comment by shilangyu** [comment]
Kinda sad that it bumps dart SDK to 2.17. do we want to enable it in all of our packages?

**Comment by bartekpacia** [comment]
Since new lints are available since Dart 2.17, I think it makes sense to require it, right?

> do we want to enable it in all of our packages?
I think that ultimately, yes.

File: packages/leancode_lint/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,22 @@
+# 1.2.0
```
Comment (shilangyu) [inline packages/leancode_lint/CHANGELOG.md:1]: I would mention bumping min dart SDK to 2.17

Comment (bartekpacia) [reply packages/leancode_lint/CHANGELOG.md:1]: Added.

Is setting minSdk to 2.17 a breaking change?

Comment (shilangyu) [reply packages/leancode_lint/CHANGELOG.md:1]: I wouldn't say so. It blocks upgrades but it does not break existing code.

**Review by shilangyu — APPROVED** [review]

**Pull request merged**